responses.py

```diff
@@ -288,9 +290,15 @@ def unbound_on_send(adapter, request, *a, **kwargs):
     def stop(self):
         self._patcher.stop()
         if self.assert_all_requests_are_fired and self._urls:
-            raise AssertionError(
+            assertion_error = AssertionError(
                 'Not all requests has been executed {0!r}'.format(
                     [(url['method'], url['url']) for url in self._urls]))
+            if self.captured_exception:
+                try:
```

Comment:
Ah, I now realised that the code is using exception chaining which is a Python 3 only feature. The result on python2 is that the original real exception will still go lost. On python 2, I think it would be better to just raise the captured exception.